ytypes/int_type_test.go

```diff
@@ -72,6 +72,40 @@ func TestValidateIntSchema(t *testing.T) {
 			schema:  &yang.Entry{Name: "string-type-schema", Type: &yang.YangType{Kind: yang.Ystring}},
 			wantErr: true,
 		},
+		{
+			desc: "bad range Min",
+			schema: &yang.Entry{
+				Name: "bad-range-schema",
+				Type: &yang.YangType{
+					Kind: yang.Yint8,
+					Range: yang.YangRange{
+						yang.YRange{
+							Min: yang.Number{
+								Kind: yang.MaxNumber,
```

Comment:
A couple of sides to this. First, the concept being represented: MaxNumber represents infinity, and to say a number must lie between +/- infinity and (same +/-) infinity is currently not permissible. Is this a meaningful statement, conceptually? I don't know but I don't think so. 
Second, what is the effect of actually allowing this in implementation. In practice, if a range were [MaxNumber, MaxNumber] there would not be any number that could fall in this range, because the values in the data tree are just regular numbers, there's no concept of yang.Number there (Max/MinNumber are types, not values). So just for that practical reason I don't think it's a good idea to allow this.

Reply:
Ah, OK -- I didn't (lazily, sorry) go and check that MaxNumber wasn't a defined actual number. If this is the case, then yes, I agree.